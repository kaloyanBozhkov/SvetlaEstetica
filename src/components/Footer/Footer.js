import React from 'react'
import PropTypes from 'prop-types'
import { Flex, Text } from '@chakra-ui/core'
import Link from 'next/link'

const Footer = ({
    linksConfig = [],
    findUsLink = { url: '', text: '' },
    brandName = '',
    brandAddress = '',
}) => {
    return (
        <Flex>
            <Flex>
                <Text>{brandName}</Text>
                <Text>{brandAddress}</Text>
                <Link as={Text} href={findUsLink.url}>
                    {findUsLink.text}
                </Link>
            </Flex>
            <Flex>
                <Text>Supporto</Text>
                {linksConfig.map(({ url, text }, key) => (
                    <Link data-test-id="link" key={key} as={Text} href={url}>
                        {text}
                    </Link>
                ))}
            </Flex>
        </Flex>
    )
}

Footer.propTypes = {
    linksConfig: PropTypes.arrayOf(
        PropTypes.shape({
            url: PropTypes.string,
            text: PropTypes.string,
        })
    ),
    findUsLink: PropTypes.shape({
        url: PropTypes.string,
        text: PropTypes.string,
    }),
    brandName: PropTypes.string,
    brandAddress: PropTypes.string,
}

export default Footer
