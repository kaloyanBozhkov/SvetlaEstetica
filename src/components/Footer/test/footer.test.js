import React from 'react'
import { shallow } from 'enzyme'
import Footer from '../footer'

describe('testing footer component', () => {
    it('should render with default props', () => {
        const wrapper = shallow(<Footer />)
        expect(wrapper).toMatchSnapshot()
    })

    it('should render footer links', () => {
        const linksConfig = [
                {
                    url: '/some-url',
                    text: 'Some Url',
                },
                {
                    url: '/another-url',
                    text: 'Another Url',
                },
            ],
            wrapper = shallow(<Footer linksConfig={linksConfig} />)

        const links = wrapper.find('[data-test-id="link"]')

        expect(links.length).toEqual(linksConfig.length)

        expect(links.at(0).prop('href')).toEqual(linksConfig[0].url)
        expect(links.at(1).prop('href')).toEqual(linksConfig[1].url)
        expect(links.at(0).prop('children')).toEqual(linksConfig[0].text)
        expect(links.at(1).prop('children')).toEqual(linksConfig[1].text)
    })
})
