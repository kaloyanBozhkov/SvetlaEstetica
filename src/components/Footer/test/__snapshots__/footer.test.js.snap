// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing footer component should render with default props 1`] = `
<Flex>
  <Flex>
    <Text />
    <Text />
    <Link
      as={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "displayName": "Text",
          "render": [Function],
        }
      }
      href=""
    />
  </Flex>
  <Flex>
    <Text>
      Supporto
    </Text>
  </Flex>
</Flex>
`;
